### PR TITLE
feat: Rework Channel, Sender, and Receiver

### PR DESCRIPTION
**Pull request by JonathanStarup**
Related to #9328

Some typing is weird because of spawn, but that is next PR

### CODE REVIEW

File: main/src/library/Channel.flix [hunk 1]
```diff
@@ -20,28 +20,28 @@ mod Channel {
     ///
```
Comment (JonathanStarup) [inline main/src/library/Channel.flix:20]: @magnus-madsen In this file is the typing change clearest - all the other code is just consequences

Comment (magnus-madsen) [reply main/src/library/Channel.flix:20]: Thanks 👍

**Comment by magnus-madsen** [comment]
Do you mind if this sits for a few days while I mull over the changes? 

I mean, I think this is the way to go, I just dislike "Global". 

Meanwhile I will also give Matt a chance to review since he stands to benefit :)

**Comment by JonathanStarup** [comment]
Sure - if its just the name you're pondering, it's easy to change

**Comment by magnus-madsen** [comment]
I think this is structurally sound. @mlutze Agree?

I am just left with two stylistic issues:

- What should the `Global` effect be called?
- What should be order we write the effects in?

That said, we should also be sure to drop `Sendable` since the new system is sound without it.

**Comment by magnus-madsen** [comment]
Brainstorming names:

- Channel (obvious) -- note that the types are called Sender and Receiver (so there is no clash)
- Comm (Communication)
- Relay
- Talk
- Message

**Comment by JonathanStarup** [comment]
> That said, we should also be sure to drop Sendable since the new system is sound without it.

Lets do in a separate PR

**Comment by magnus-madsen** [comment]
I propose:

- We call the effect `Chan` -- this does not clash with any existing name.
- We always write the effect set as a concrete set in this order: `{Chan, NonDet, IO}` (most specific to least specific)

**Comment by magnus-madsen** [comment]
I would merge this.

**Comment by JonathanStarup** [comment]
@magnus-madsen ready

**Comment by magnus-madsen** [comment]
@mlutze Can you take a quick look? Then I will follow.

**Comment by magnus-madsen** [comment]
@JonathanStarup Before I can merge this, I also need a PR that updates doc.flix.dev

**Comment by JonathanStarup** [comment]
done

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Next: 

- Fix sending-and-receiving-on-channels.flix
- Sendable is dead
- Type rule of spawn

**Pull request merged**